patches/minecraft/net/minecraft/entity/item/EntityMinecart.java.patch

```diff
@@ -254,9 +254,16 @@
                              {
                                  p_70108_1_.field_70159_w *= 0.20000000298023224D;
                                  p_70108_1_.field_70179_y *= 0.20000000298023224D;
-@@ -1015,6 +1047,229 @@
+@@ -1014,7 +1046,236 @@
+     {
          this.func_184212_Q().func_187227_b(field_184270_f, Boolean.valueOf(p_94096_1_));
      }
++    
++    @Override
++    public boolean func_184230_a(EntityPlayer player, net.minecraft.util.EnumHand hand)
```

Comment:
How can we know if the subclasses always call `super`?

Reply:
It's good practice to call `super`, and modded carts that do not call `super` for whatever reason should fire the event themselves to be proper.
Any modded cart that was not calling `super` before will not be broken more by this change.

Reply:
What if the event is fired twice?

Reply:
Valid concern... that I would like an answer to @MineMaarten

Reply:
The event will be fired twice if the current implementation of modded EntityMinecarts is along the lines of

```java
@Override
    public boolean processInitialInteract(EntityPlayer player, EnumHand hand)
    {
          if(!super.processInitialInteract(player, hand){
                 return MinecraftForge.EVENT_BUS.post(new net.minecraftforge.event.entity.minecart.MinecartInteractEvent(this, player, hand));
          }
    }
```

So when the event is not canceled from the super, then the subscribers will be called twice, of which the effects are highly dependent on the implementation.

However, I would argue this should not be a problem, because we are talking about interacting an entity, something you could do in quick succession manually by just right clicking the cart twice. A subscriber expecting the game to go back in a 'stable state' via another way before allowing the event to fire again feels wrong to me, I would certainly not expect crashes because of this.

I did think of an example where you would get unwanted side-effects though. Let's say you have an item you can 'apply' to any minecart, multiple times. By right clicking a cart with the item, it uses one of the items from the stack. So by firing the event multiple times, with one right click multiple items would be used.

Hmm another example would be RailCraft's cart linking, which currently will toggle the linking/unlinking twice in the case of firing the event twice:
https://github.com/Railcraft/Railcraft/blob/aed27a85a118b6aed7818974f3805875c47b4e16/src/main/java/mods/railcraft/common/items/CrowbarHandler.java#L78-L87

I haven't seen this implementation (snippet at the top) though. Probably because the super method currently just returns false so people don't bother. So yes, this PR won't fix the cases where the method is overriden, but the super is not called, in which case they already fire the event themselves (for example https://github.com/Railcraft/Railcraft/blob/9966ac111256c04275ca966cd6b28e2918e1ae45/src/main/java/mods/railcraft/common/carts/CartBase.java#L56-L58 ), but it will fix the plenty cases where a basic subclass is made from EntityMinecart which does not have interaction and therefore does not override the method (including the vanilla EntityMinecartTNT and EntityMinecartMobSpawner, which would've been patched, currently).

Reply:
Railcraft is still in 1.10.2. I agree that this change is going to be safe.

Reply:
Honestly with RailCraft being in 1.10.2, im inclined to drop all our EntityMinecart modifications as its a major PITA. And it was done specifically for Railcraft.
I'll pull this in, but I think those who add custom minecarts SHOULD go through and re-evaluate things for 1.13+